### PR TITLE
Feature#5

### PR DESCRIPTION
**Pull request by jtparks**
Working Version of Battleship

### CODE REVIEW

File: src/main/java/controllers/ApplicationController.java [hunk 1]
```diff
@@ -33,6 +33,7 @@ public Result placeShip(Context context, PlacementGameAction g) {
     public Result attack(Context context, AttackGameAction g) {
         Game game = g.getGame();
         boolean result = game.attack(g.getActionRow(), g.getActionColumn());
+        System.out.println("Result attacked in " + result);
```
Comment (croysdar) [inline src/main/java/controllers/ApplicationController.java]: This should be removed

Comment (jtparks) [reply src/main/java/controllers/ApplicationController.java]: Done!

File: src/main/java/cs361/battleships/models/Game.java [hunk 2]
```diff
@@ -36,7 +34,9 @@ public boolean placeShip(Ship ship, int x, char y, boolean isVertical) {
 	DO NOT change the signature of this method. It is used by the grading scripts.
 	 */
     public boolean attack(int x, char  y) {
+        System.out.println("Attacking Player");
```
Comment (croysdar) [inline src/main/java/cs361/battleships/models/Game.java]: Should be removed

File: src/main/java/cs361/battleships/models/Game.java [hunk 3]
```diff
@@ -36,7 +34,9 @@ public boolean placeShip(Ship ship, int x, char y, boolean isVertical) {
...
         Result playerAttack = opponentsBoard.attack(x, y);
+        System.out.println("Player Attack" + playerAttack);
```
Comment (croysdar) [inline src/main/java/cs361/battleships/models/Game.java]: Should be removed

File: src/main/java/cs361/battleships/models/Game.java [hunk 4]
```diff
@@ -45,8 +45,12 @@ public boolean attack(int x, char  y) {
         do {
             // AI does random attacks, so it might attack the same spot twice
             // let it try until it gets it right
+            System.out.println("Attacking Opponent");
```
Comment (croysdar) [inline src/main/java/cs361/battleships/models/Game.java]: Should be removed

File: src/main/java/cs361/battleships/models/Game.java [hunk 5]
```diff
@@ -45,8 +45,12 @@ public boolean attack(int x, char  y) {
...
             opponentAttackResult = playersBoard.attack(randRow(), randCol());
-        } while(opponentAttackResult.getResult() != INVALID);
+            System.out.println("Opponent Attack" + opponentAttackResult);
```
Comment (croysdar) [inline src/main/java/cs361/battleships/models/Game.java]: Should be removed

File: src/main/java/cs361/battleships/models/Game.java [hunk 6]
```diff
@@ -45,8 +45,12 @@ public boolean attack(int x, char  y) {
...
+            System.out.println("Row: " + opponentAttackResult.getLocation().getRow() + " Col: " + opponentAttackResult.getLocation().getColumn());
```
Comment (croysdar) [inline src/main/java/cs361/battleships/models/Game.java]: Should be removed

File: src/main/java/cs361/battleships/models/Board.java [hunk 7]
```diff
@@ -87,10 +92,70 @@ public boolean placeShip(Ship ship, int x, char y, boolean isVertical) {
 	DO NOT change the signature of this method. It is used by the grading scripts.
 	 */
 	public Result attack(int x, char y) {
-		//TODO Implement
-		return null;
+		Result aResult = new Result();
```
Comment (AmbiguousCat) [inline src/main/java/cs361/battleships/models/Board.java]: Please add comments to your code. :)

**Comment by loganintech** [comment]
This will close #7, #8, and #5.

**Pull request merged**